text/0120-metric-trace-log-conversion.md

```diff
@@ -0,0 +1,193 @@
+# Metric and Trace log conversion
+
+Specifying the structure of trace and metric data in logs.
+
+## Motivation
+
+There are multiple requests and issues related to converting metric/tracing data into logs:
+
+* [Issue 398](https://github.com/open-telemetry/opentelemetry-specification/issues/398)
+* [Issue 617](https://github.com/open-telemetry/opentelemetry-specification/issues/617)
+
+The [aim here](https://gitter.im/open-telemetry/logs?at=5ee284f2ef5c1c28f0194a89) is to create a standard method to convert a trace or metric into a log for Otel exporters to reduce confusion and increase compatibility.
```

Comment:
Good points. After some other feedback I got, I'll make the intention more clear, and also shift the goals a bit. The updates will include multiple export formats depending on intent and purpose, instead of only having a single way of doing things. Also, thank you for making me aware of that prior work.

Reply:
@avik-so do you intend to update this OTEP?

Reply:
I do.
Still doing research on my part. I'm happy to get assistance.

Reply:
As for Metrics, see https://github.com/open-telemetry/opentelemetry-proto/pull/181.

This PR amounts to a (rejected) proposal for OTLP. The problem with https://github.com/open-telemetry/opentelemetry-proto/pull/181 is that it was better at describing the input of an aggregation, not the output of an aggregation.

I think if we're logging, the idea is to log **raw** metric events (i.e., no aggregation). The proposal in that PR has a MetricDescriptor with these fields:

- Temporality (Cumulative, Delta)
- Structure (Adding Monotonic, Adding, Grouping)
- Continuity (Continuous, Snapshot)
= 12 combinations

These 12 combinations can be thought of as six instruments crossed with cumulative/delta, which completely and accurately describes the current OTel API metric event.

That led to very lengthy explanations of _what it means_ for various kinds aggregation. If we're only considering raw metric events, then this may be a good protocol.

### Technical notes

The problem with https://github.com/open-telemetry/opentelemetry-proto/pull/181 may have been predicted by #88 and #93, where more than six potential kinds of instrument were described and then some of them were rejected.  See also https://github.com/open-telemetry/opentelemetry-proto/pull/199--I believe that a metric descriptor with Structure=Grouping and Temporality=Delta could also be described as the hypothetical "Interval" temporality mentioned there.

There is also a question of whether you can derive `system.cpu.utilization` or `process.cpu.utilization` correctly with whatever logging format is proposed: https://github.com/open-telemetry/opentelemetry-specification/issues/819